Las/Core/Credentials.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+using IniParser;
+using IniParser.Model;
+
+namespace Lucidtech.Las.Core
+{
+    /// <summary>
+    /// Used to fetch and store credentials. One of 3 conditions must be met to successfully create credentials.
+    /// 
+    /// 1. credentials_path is provided
+    /// 2. access_key_id, secret_access_key and api_key is provided
+    /// 3. credentials is located in default path ~/.lucidtech/credentials.cfg
+    /// 
+    /// </summary>
+    public class Credentials
+    {
+        public string AccessKeyId {get;}
+        public string SecretAccessKey {get;}
+        public string ApiKey{get;}
+        
+        /// <summary>
+        /// The Credentials constructor, will create all the necessary credentials.
+        /// 
+        /// 1. credentials_path is provided
+        /// 2. access_key_id, secret_access_key and api_key is provided
+        /// 3. credentials is located in default path ~/.lucidtech/credentials.cfg
+        /// 
+        /// </summary>
+        /// <param name="credentialsPath"> Path to credentials file</param>
+        /// <param name="accessKeyId">Access key Id</param>
+        /// <param name="secretAccessKey">Secret Access Key</param>
+        /// <param name="apiKey">API key</param>
+        /// <exception cref="ArgumentException"></exception>
+        public Credentials(string credentialsPath = "", string accessKeyId = "", string secretAccessKey = "",
+            string apiKey = "")
+        {
+            string[] input = 
+            {
+                accessKeyId,
+                secretAccessKey,
+                apiKey,
+                credentialsPath
+            };
+            
+            string[] credentials = input.Take(3).ToArray(); 
+            
+            if (input.Any(s => string.IsNullOrEmpty(s)))
+            {
+                if (string.IsNullOrEmpty(credentialsPath))
+                {
+                    string path = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)?"%UserProfile%\\.lucidtech\\credentials.cfg":"%HOME%/.lucidtech/credentials.cfg";
```

Comment:
Is there not a library that handles cross-platform paths in .NET? Having to manually specify the OS-dependent equivalent of home folder seems prone to errors.
Also missing whitespaces around "?" and ":"